components/FutureGraph.tsx

```diff
@@ -0,0 +1,318 @@
+          new Date(data.forecast_values[0].target_datetime_utc)
+        );
+        const endTime = formatter.format(
+          new Date(
+            data.forecast_values[numForecastValues - 1].target_datetime_utc
+          )
+        );
+
+        return (
+          <div className="flex flex-row justify-between">
+            <p className="text-white text-xs font-medium ml-6">{startTime}</p>
+            <p className="text-white text-xs font-medium mr-6">{endTime}</p>
+          </div>
+        );
+      }
+    }
+
+    return null;
+  };
+
+  const solarIncreasingText = (formattedDate: string) => {
+    return (
+      <div className="flex flex-row justify-center mt-2">
+        <UpArrowIcon />
+        <p className="text-white text-sm font-normal ml-2">
+          Solar activity is increasing until {formattedDate}
+        </p>
+      </div>
+    );
+  };
+
+  const solarDecreasingText = (formattedDate: string) => {
+    return (
+      <div className="flex flex-row justify-center mt-2">
+        <DownArrowIcon />
+        <p className="text-white text-sm font-normal ml-2">
+          Solar activity is decreasing until {formattedDate}
+        </p>
+      </div>
+    );
+  };
+
+  /**
+   * Calculates the next peak/trough starting from the current time
+   * and returns text indicating increasing/decreasing solar activity
+   */
+  const getSolarActivityText = () => {
+    if (data) {
+      let currIndex = getCurrentTimeForecastIndex() + 1;
+
+      while (currIndex < data.forecast_values.length) {
+        const currentExpectedGenerationKW =
+          data.forecast_values[currIndex].expected_generation_kw;
+        const previousExpectedGenerationKW =
+          data.forecast_values[currIndex - 1].expected_generation_kw;
+        const formattedDate = formatter.format(
+          new Date(data.forecast_values[currIndex].target_datetime_utc)
+        );
+
+        if (currIndex === data.forecast_values.length - 1) {
+          if (currentExpectedGenerationKW > previousExpectedGenerationKW) {
+            return solarIncreasingText(formattedDate);
+          } else if (
+            currentExpectedGenerationKW < previousExpectedGenerationKW
+          ) {
+            return solarDecreasingText(formattedDate);
+          }
+        } else {
+          const nextExpectedGenerationKW =
+            data.forecast_values[currIndex + 1].expected_generation_kw;
+
+          if (
+            currentExpectedGenerationKW > previousExpectedGenerationKW &&
+            currentExpectedGenerationKW > nextExpectedGenerationKW
+          ) {
+            return solarIncreasingText(formattedDate);
+          } else if (
+            currentExpectedGenerationKW < previousExpectedGenerationKW &&
+            currentExpectedGenerationKW < nextExpectedGenerationKW
+          ) {
+            return solarDecreasingText(formattedDate);
+          }
+        }
+        currIndex += 1;
+      }
+    }
+
+    return '';
+  };
```

Comment:
Consider abstracting this out into utils or another suitable file—this file getting pretty long, and we might want to use this function in other future modules for our dashboard!